src/services/room-service.js

```diff
@@ -1,21 +1,76 @@
-import RoomStore from '../stores/room-store.js';
+import { add } from 'date-fns';
+import httpErrors from 'http-errors';
+import Logger from '../common/logger.js';
+import UserService from './user-service.js';
 import uniqueId from '../utils/unique-id.js';
+import RoomStore from '../stores/room-store.js';
+import { ROOM_ACCESS_LEVEL } from '../common/constants.js';
+import RoomInvitationStore from '../stores/room-invitation-store.js';
+
+const { BadRequest, NotFound } = httpErrors;
+
+const logger = new Logger(import.meta.url);
+
+const PENDING_ROOM_INVITATION_EXPIRATION_TIMESPAN = { days: 7 };
 
 export default class RoomService {
-  static get inject() { return [RoomStore]; }
+  static get inject() { return [RoomStore, RoomInvitationStore, UserService]; }
 
-  constructor(roomsStore) {
-    this.roomsStore = roomsStore;
+  constructor(roomStore, roomInvitationStore, userService) {
+    this.roomStore = roomStore;
+    this.roomInvitationStore = roomInvitationStore;
+    this.userService = userService;
   }
 
-  async createRoom(room) {
+  async createRoom({ name, access, user }) {
     const newRoom = {
       _id: uniqueId.create(),
-      ...room,
+      name,
+      access,
+      owner: user._id,
       members: []
     };
 
-    await this.roomsStore.save(newRoom);
+    await this.roomStore.save(newRoom);
     return newRoom;
   }
+
+  async findOwnedRoomById({ roomId, ownerId }) {
```

Comment:
What do you think about moving this logic directly in `createOrUpdateInvitation` and throwing a separate cleare error if `room.owner !== ownerId` ?

Reply:
I had this first, then moved it here cause we're gonna need that logic with the owner check more often in the future, but I agree that this is premature optimization. What do you think about throwing the "NotFound" error directly here instead of in the consuming code, then we ensure consistent behavior?

Reply:
That's also a good option, yes.
We could still throw 2 errors, one for room not found and one for owner not matching 🤔 I think it would be a useful distinction.

Reply:
Actually, let's just log that but throw 404 in both cases, if a user is not a room owner it's non of their business if the room exists or not (thus showing "does not exist").

Reply:
That's true.

Reply:
Changed!